astropy/io/fits/connect.py

```diff
@@ -141,20 +141,23 @@ def read_table_fits(input, hdu=None, astropy_native=False):
     # Convert to an astropy.table.Table object
     # Note: in future, it may make more sense to do this column-by-column,
     # rather than via the structured array.
-    t = Table(table.data, masked=masked)
+    t = Table(masked=masked)
+
+    for col in table.data._coldefs:
```

Comment:
It should be somewhat more efficient to make a list of `Column`s and create `t` from that.